File: algorithms/0009.palindrome_number/solution.go
package algorithms

// 9. Palindrome Number (Easy)
// https://leetcode.com/problems/palindrome-number/

// Modulo and division solution
func isPalindrome(number int) bool {
	if number < 0 {
		return false
	}

	originalNumber := number
	reversedNumber := 0

	for number > 0 {
		remainder := number % 10
		reversedNumber = (reversedNumber * 10) + remainder
		number /= 10
	}

	return reversedNumber == originalNumber
}

File: algorithms/0009.palindrome_number/solution.ts
// 9. Palindrome Number (Easy)
// https://leetcode.com/problems/palindrome-number/

// Modulo and division solution
const isPalindrome = (num: number): boolean => {
  if (num < 0) return false

  const originalNum = num
  let reversedNum = 0

  while (num > 0) {
    const remainder = num % 10
    reversedNum = reversedNum * 10 + remainder
    num = Math.floor(num / 10)
  }

  return reversedNum === originalNum
}
